### ISSUE TITLE
Can't make this project work, on Cargo v1.76.0

### ISSUE THREAD

**Issue by Naereen** [maekawatoshiki/rcaml#1]
Hi,
I'm very interested in this project, and wanted to try it out!
I downloaded it and tried to `cargo run example/test.ml` it using a hand-written tiny `test.ml` file.
I'm getting the following error message:

```bash
$ cargo run example/test.ml
Running `target/debug/rcaml example/test.ml`
thread 'main' panicked at src/main.rs:41:36:
error: Alt
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I'm *not at all* an expert Rust developper, so maybe you could help me to know what to do now or what to fix?
Thanks. Regards from France
@Naereen

**Comment by maekawatoshiki** [comment]
Hi.
Since the project is from several years ago, I'm not sure I can help you.

The error looks it's raised from parsing phase. Can you provide your `test.ml`, which perhaps simply contains some unsupported syntax?

**Comment by Naereen** [comment]
Hi again,
Thanks for an impressively quick response!
I've tried each test files, and couldn't run any of them.

**Comment by Naereen** [comment]
Each of them fails, with the same error message.

**Comment by maekawatoshiki** [comment]
Are the 'test files' referring to the files under `./example` ?

**Comment by Naereen** [comment]
Yes. I don't see any errors when I simply run `cargo run` (it seems to work fine), but I have the same error when loading any of these files:
```
example/tuple.ml
example/mandelbrot.ml
example/fibo.ml
example/adder.ml
example/test.ml
```

**Comment by maekawatoshiki** [comment]
Hmm, it seems, in the branch `poly`, you can run `./example/fibo.ml`

**Comment by Naereen** [comment]
I'm trying, and I'm getting the following error, for `cargo run` as well as `cargo run ./example/fibo.ml`:
```
Running `target/debug/rcaml example/fibo.ml`
Erreur de segmentation (core dumped)
```

**Comment by Naereen** [comment]
(segmentation fault)

**Comment by maekawatoshiki** [comment]
Make sure you correctly installed LLVM 3.9.
For reproducibility, recommend to try it inside a container.

**Comment by Naereen** [comment]
I can't install LLVM 3.9 from my `apt` packages, on Ubuntu 22.04 on this laptop.
Apparently it's not trivial to install older versions.

I'll try running from a container, as soon as I have a solid Internet connexion, thanks for the tip!
